ruby/red-arrow/lib/arrow/raw-table-converter.rb

```diff
@@ -35,7 +35,11 @@ def convert
         fields = []
         @values = []
         @raw_table.each do |name, array|
-          array = ArrayBuilder.build(array) if array.is_a?(::Array)
+          if array.respond_to?(:to_arrow_array)
+            array = array.to_arrow_array
+          elsif array.is_a?(::Array)
```

Comment:
Do you think how to handle objects that respond to `to_ary` in addition to normal arrays?

Reply:
It's a good idea.
I've added support for `#to_ary`.

Reply:
Ok. LGTM!